app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,49 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isActive/isUsed to true, while deactivating the old's
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Deactivate all terminals
+      _.each($scope.containerTerminals, function(term) {
+        term.isActive = false;
+      });
+
+      newTerm.value.isActive = true;
+      newTerm.value.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(containerStatuses, existingTerms) {
+      existingTerms = existingTerms || {};
+
+      var firstRun = !!existingTerms;
+
+      _.each(containerStatuses, function(status, i) {
```

Comment:
Disable the option for containers that aren't running, `!status.state.running`?

https://github.com/angular-ui/ui-select/wiki/ui-select-choices#example-disable-choices